File: CPP/No190.cc
/**
 * Created by Xiaozhong on 2020/8/19.
 * Copyright (c) 2020/8/19 Xiaozhong. All rights reserved.
 */
#include "iostream"

using namespace std;

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ans = 0, power = 31;
        while (n != 0) {
            ans += (n & 1) << power;
            power--;
            n = n >> 1;
        }
        return ans;
    }
};

int main() {
    Solution s;
    cout << s.reverseBits(uint32_t(43261596)) << endl;
}

File: Java/src/No190.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No190 {
    public int reverseBits(int n) {
        int ans = 0;
        for (int i = 0; i < 32 && n != 0; i++) {
            ans |= (n & 1) << (31 - i);
            n >>= 1;
        }
        return ans;
    }
}
